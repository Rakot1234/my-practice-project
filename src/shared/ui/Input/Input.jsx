import React, { Component } from 'react';
import './Input.scss';
import PropTypes from 'prop-types';
import cx from 'classnames';

class Input extends Component {
    static propTypes = {
        className: PropTypes.string,
        placeholder: PropTypes.string,
        title: PropTypes.string,
        isRequired: PropTypes.bool,
        type: PropTypes.oneOf([
            'checkbox',
            'email',
            'radio',
            'search',
            'submit',
            'text',
            'tel'
        ]),
    };

    static defaultProps = {
        placeholder: '',
        type: 'text'
    };

    render() {
        const { placeholder, className, type, isRequired, title } = this.props;

        return (
            <label className={cx('input')}>
                {title && <span className={cx('input__title')}>{title}</span>}
                <input
                    type={type}
                    className={cx('input__input', className)}
                    placeholder={placeholder}
                    required={isRequired}
                />
            </label>
        );
    }
};

export default Input;
