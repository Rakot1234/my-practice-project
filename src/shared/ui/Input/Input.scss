@import '../../../styles/variables.scss';

$block: input;

.#{$block} {
    &:focus {
        outline: none;
    }

    &[type="search"]::-ms-clear,
    &[type="search"]::-ms-reveal
    { 
        display: none;
        width : 0;
        height: 0;
    }

    &[type="search"]::-webkit-search-decoration,
    &[type="search"]::-webkit-search-cancel-button,
    &[type="search"]::-webkit-search-results-button,
    &[type="search"]::-webkit-search-results-decoration
    {
        display: none;
    }

    &__label {
        display: flex;
        align-items: center;
        height: 100%;
    }

    &__label_justify {
        &_start {
            justify-content: flex-start;
        }

        &_center {
            justify-content: center;
        }

        &_end {
            justify-content: flex-end;
        }

        &_space-around {
            justify-content: space-around;
        }

        &_space-between {
            justify-content: space-between;
        }
    }
}