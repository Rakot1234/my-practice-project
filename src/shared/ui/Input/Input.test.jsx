import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('<Input />', () => {
    it('Renders input component', () => {
        const wrapper = shallow(<Input />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Shoud be changed', () => {
        const onChange = jest.fn();
        const wrapper = shallow(<Input onChange={onChange} />);
        wrapper.find('.input__input').simulate('change');
        expect(onChange.mock.calls).toHaveLength(1);
    });
});