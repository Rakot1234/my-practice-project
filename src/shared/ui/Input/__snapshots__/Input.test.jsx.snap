// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Renders input component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    isDisabled={false}
    itemsJustify="start"
    placeholder=""
    type="text"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <InputElement
          className="input"
          disabled={false}
          placeholder=""
          readOnly={true}
          type="text"
          value=""
        />,
      ],
      "className": "input__label input__label_justify_start",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "input",
          "disabled": false,
          "mask": undefined,
          "name": undefined,
          "onChange": undefined,
          "placeholder": "",
          "readOnly": true,
          "required": undefined,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <InputElement
            className="input"
            disabled={false}
            placeholder=""
            readOnly={true}
            type="text"
            value=""
          />,
        ],
        "className": "input__label input__label_justify_start",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "input",
            "disabled": false,
            "mask": undefined,
            "name": undefined,
            "onChange": undefined,
            "placeholder": "",
            "readOnly": true,
            "required": undefined,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
